glib/src/auto/match_info.rs

```diff
@@ -0,0 +1,141 @@
+
+    #[doc(alias = "g_match_info_fetch_all")]
+    pub fn fetch_all(&self) -> Vec<crate::GString> {
+        unsafe {
+            FromGlibPtrContainer::from_glib_full(ffi::g_match_info_fetch_all(self.to_glib_none().0))
+        }
+    }
+
+    #[doc(alias = "g_match_info_fetch_named")]
+    pub fn fetch_named(&self, name: &str) -> Option<crate::GString> {
+        unsafe {
+            from_glib_full(ffi::g_match_info_fetch_named(
+                self.to_glib_none().0,
+                name.to_glib_none().0,
+            ))
+        }
+    }
+
+    #[doc(alias = "g_match_info_fetch_named_pos")]
+    pub fn fetch_named_pos(&self, name: &str) -> Option<(i32, i32)> {
+        unsafe {
+            let mut start_pos = mem::MaybeUninit::uninit();
+            let mut end_pos = mem::MaybeUninit::uninit();
+            let ret = from_glib(ffi::g_match_info_fetch_named_pos(
+                self.to_glib_none().0,
+                name.to_glib_none().0,
+                start_pos.as_mut_ptr(),
+                end_pos.as_mut_ptr(),
+            ));
+            if ret {
+                Some((start_pos.assume_init(), end_pos.assume_init()))
+            } else {
+                None
+            }
+        }
+    }
+
+    #[doc(alias = "g_match_info_fetch_pos")]
+    pub fn fetch_pos(&self, match_num: i32) -> Option<(i32, i32)> {
+        unsafe {
+            let mut start_pos = mem::MaybeUninit::uninit();
+            let mut end_pos = mem::MaybeUninit::uninit();
+            let ret = from_glib(ffi::g_match_info_fetch_pos(
+                self.to_glib_none().0,
+                match_num,
+                start_pos.as_mut_ptr(),
+                end_pos.as_mut_ptr(),
+            ));
+            if ret {
+                Some((start_pos.assume_init(), end_pos.assume_init()))
+            } else {
+                None
+            }
+        }
+    }
+
+    #[doc(alias = "g_match_info_get_match_count")]
+    #[doc(alias = "get_match_count")]
+    pub fn match_count(&self) -> i32 {
+        unsafe { ffi::g_match_info_get_match_count(self.to_glib_none().0) }
+    }
+
+    #[doc(alias = "g_match_info_get_regex")]
+    #[doc(alias = "get_regex")]
+    pub fn regex(&self) -> Regex {
+        unsafe { from_glib_none(ffi::g_match_info_get_regex(self.to_glib_none().0)) }
+    }
+
+    #[doc(alias = "g_match_info_get_string")]
+    #[doc(alias = "get_string")]
+    pub fn string(&self) -> crate::GString {
+        unsafe { from_glib_none(ffi::g_match_info_get_string(self.to_glib_none().0)) }
+    }
+
+    #[doc(alias = "g_match_info_is_partial_match")]
+    pub fn is_partial_match(&self) -> bool {
+        unsafe { from_glib(ffi::g_match_info_is_partial_match(self.to_glib_none().0)) }
+    }
+
+    #[doc(alias = "g_match_info_matches")]
+    pub fn matches(&self) -> bool {
+        unsafe { from_glib(ffi::g_match_info_matches(self.to_glib_none().0)) }
+    }
+
+    #[doc(alias = "g_match_info_next")]
+    pub fn next(&self) -> Result<(), crate::Error> {
```

Comment:
Should there maybe be an `Iterator` impl that works over this?

Reply:
Looking at this in a bit more detail, the `g_match_info_next` method doesn't return a new `MatchInfo`, it modifies the existing one in place, so it may be quite difficult to implement the `Iterator` trait over the top of this? I assume we'd need some kind of copy mechanics on the underlying struct?

Reply:
Let's not worry about that then for now :)